packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts

```diff
@@ -72,40 +74,53 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
   }
 
   const conditionalRetry = function* (id: string) {
-    const { RETRY_AMOUNT, SECONDS } = POLLING
-    const data = yield retry(RETRY_AMOUNT, SECONDS * 1000, transferAccountState, id)
-    return data
-  }
+    const MAX_TRIES = Array(6)
+    let ITERATION = 1
 
-  const fetchBankTransferUpdate = function* ({
-    payload
-  }: ReturnType<typeof A.fetchBankTransferUpdate>) {
-    try {
-      const account = payload
+    const decay = (multiplier: number) => {
+      return multiplier * 1000
+    }
 
-      let bankId
-      let attributes
-      const bankCredentials = S.getBankCredentials(yield select()).getOrElse({} as OBType)
-      const fastLink = S.getFastLink(yield select()).getOrElse({} as FastLinkType)
+    while (MAX_TRIES.length) {
+      try {
+        return yield call(transferAccountState, id)
+      } catch (error) {
+        yield delay(decay(ITERATION))
+        ITERATION += ITERATION * 2
+        MAX_TRIES.pop()
+      }
+    }
+  }
 
-      if (typeof account === 'string' && bankCredentials) {
+  const getBankUpdatePayload = function* (
+    bankCredentials: BankCredentialsType,
+    account: string | PlaidAccountType | YodleeAccountType
+  ) {
+    switch (true) {
+      case typeof account === 'string' && bankCredentials:
```

Comment:
what else can the `account` be?

Reply:
`account` is what wallet receives from the iframe's postMessage so it's different for each provider (plaid, yapily, yodlee). Now that i'm thinking about it though, if I just standardize that I would be able to get rid of some duplicate code. brb! 🏃🏻